shell/platform/android/android_surface_software.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/android/android_surface_software.h"
+#include "flutter/common/threads.h"
+#include "flutter/fml/platform/android/jni_weak_ref.h"
+#include "flutter/fml/platform/android/scoped_java_ref.h"
+#include "flutter/shell/platform/android/platform_view_android_jni.h"
+
+#include <memory>
+#include <vector>
+
+#include "flutter/fml/trace_event.h"
+#include "lib/ftl/logging.h"
+
+namespace shell {
+
+AndroidSurfaceSoftware::AndroidSurfaceSoftware() : AndroidSurface() {}
+
+AndroidSurfaceSoftware::~AndroidSurfaceSoftware() = default;
+
+bool AndroidSurfaceSoftware::IsValid() const {
+  return true;
+}
+
+bool AndroidSurfaceSoftware::ResourceContextMakeCurrent() {
+  // Resource Context always not available on software backend.
+  return false;
+}
+
+std::unique_ptr<Surface> AndroidSurfaceSoftware::CreateGPUSurface() {
+  if (!IsValid()) {
+    return nullptr;
+  }
+
+  auto surface = std::make_unique<GPUSurfaceSoftware>(this);
+
+  if (!surface->IsValid()) {
+    return nullptr;
+  }
+
+  return surface;
+}
+
+sk_sp<SkSurface> AndroidSurfaceSoftware::AcquireBackingStore(
+    const SkISize& size) {
+  TRACE_EVENT0("flutter", "AndroidSurfaceSoftware::AcquireBackingStore");
+  if (!IsValid()) {
+    return nullptr;
+  }
+
+  if (sk_surface_ != nullptr &&
+      SkISize::Make(sk_surface_->width(), sk_surface_->height()) == size) {
+    // The old and new surface sizes are the same. Nothing to do here.
+    return sk_surface_;
+  }
+
+  sk_surface_ = SkSurface::MakeRasterN32Premul(
+      size.fWidth, size.fHeight, nullptr /* SkSurfaceProps as out */);
+  return sk_surface_;
+}
+
+bool AndroidSurfaceSoftware::PresentBackingStore(
```

Comment:
This can wait for a future change, but I'd be interested in looking at whether we can post the new bitmap contents to the ANativeWindow backing the FlutterView here.  This would avoid the need to call into Java and then switch over to the platform thread for painting.